ops/pipeline/build-r-docs-impl.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+if [[ $# -ne 1 ]]
+then
+  echo "Usage: $0 [branch name]"
+  exit 1
+fi
+
+set -euo pipefail
+
+branch_name=$1
+
+echo "R_LIBS_USER: ${R_LIBS_USER}"
+
+gosu root chown -R $UID:$GROUPS ${R_LIBS_USER}
+
+cd R-package
+MAKEFLAGS=-j$(nproc) Rscript ./tests/helper_scripts/install_deps.R
```

Comment:
Some of these R dependencies have system dependencies - see here for example: https://github.com/dmlc/xgboost/pull/11101/files#diff-17796461dde560588f341bc929041c6576bc9ffffa049121918f7d950ae0ee18